server/tm_cmds.go

```diff
@@ -123,3 +127,67 @@ func VersionCmd() *cobra.Command {
 		},
 	}
 }
+
+// makeKeyMigrateCmd is ported from tendermint's key-migrate command, but
+// uses the SDK's own server.Context.
+// ref: https://github.com/tendermint/tendermint/blob/master/UPGRADING.md#database-key-format-changes
+func makeKeyMigrateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "key-migrate",
```

Comment:
> Is there a way of adding this inside the upgrade handler somehow? Has this happened before (the fact that we need to run another separate command for an upgrade)?

@facundomedica In the upgrade handler, no. See the original issue, we need to run the key migration before starting tendermint, so way before the upgrade handler is called.

Reply:
Alternative solution, the SDK's `start` command is a combination of TM's `key-migrate` + `start`.
`key-migrate` is idempotent, so it can be run multiple times.

However, if we decide to show logs, as we do in this PR and as it's done in tendermint, then those logs will be shown on each startup:
```
11:55AM INF beginning a key migration dbctx=blockstore num=1 total=6
11:55AM INF beginning a key migration dbctx=state num=2 total=6
11:55AM INF beginning a key migration dbctx=peerstore num=3 total=6
11:55AM INF beginning a key migration dbctx=tx_index num=4 total=6
11:55AM INF beginning a key migration dbctx=evidence num=5 total=6
11:55AM INF beginning a key migration dbctx=light num=6 total=6
```

Any opinions on separate `key-migrate` and `start` commands VS one `start` command which runs `key-migrate` under the hood?